src/lib/storage/table.cpp

```diff
@@ -17,57 +17,112 @@
-  // Implementation goes here
-  return ColumnID{0};
+  const auto find_result_iter =
+      std::find_if(_columns.begin(), _columns.end(), [&column_name](const Column& c) { return c.name == column_name; });
+  if (find_result_iter == _columns.end()) {
+    throw std::invalid_argument("Column name does not exists.");
+  }
+  const auto find_index = static_cast<uint16_t>(std::distance(_columns.begin(), find_result_iter));
+  return ColumnID{find_index};
 }
 
-ChunkOffset Table::target_chunk_size() const {
-  // Implementation goes here
-  return 0;
-}
+ChunkOffset Table::target_chunk_size() const { return _target_chunk_size; }
 
-const std::vector<std::string>& Table::column_names() const {
-  throw std::runtime_error("Implement Table::column_names()");
+const std::vector<std::string> Table::column_names() const {
+  auto column_names = std::vector<std::string>();
+  column_names.reserve(_columns.size());
+  for (const auto& column : _columns) {
+    column_names.emplace_back(column.name);
+  }
+  return column_names;
 }
 
 const std::string& Table::column_name(const ColumnID column_id) const {
-  throw std::runtime_error("Implement Table::column_name");
+  DebugAssert(column_id < column_count(), "\"column_id\" is out of bounds.");
+  return _columns[column_id].name;
 }
 
 const std::string& Table::column_type(const ColumnID column_id) const {
-  throw std::runtime_error("Implement Table::column_type");
+  DebugAssert(column_id < column_count(), "\"column_id\" is out of bounds.");
+  return _columns[column_id].type;
 }
 
-Chunk& Table::get_chunk(ChunkID chunk_id) { throw std::runtime_error("Implement Table::get_chunk"); }
+Chunk& Table::get_chunk(ChunkID chunk_id) {
+  DebugAssert(chunk_id < column_count(), "\"chunk_id\" is out of bounds.");
+  return *_chunks[chunk_id];
+}
+
+const Chunk& Table::get_chunk(ChunkID chunk_id) const {
+  DebugAssert(chunk_id < column_count(), "\"column_id\" is out of bounds.");
+  return *_chunks[chunk_id];
+}
 
-const Chunk& Table::get_chunk(ChunkID chunk_id) const { throw std::runtime_error("Implement Table::get_chunk"); }
+void Table::_append_new_chunk() {
+  auto new_chunk = std::make_unique<Chunk>();
+
+  for (const auto& column : _columns) {
+    // append existing columns as segments to new chunk
+    new_chunk->add_segment(_create_value_segment_for_type(column.type));
+  }
+  _chunks.emplace_back(std::move(new_chunk));
+}
+
+std::shared_ptr<BaseSegment> Table::_create_value_segment_for_type(const std::string& type) {
+  auto new_segment = std::shared_ptr<BaseSegment>{};
+  resolve_data_type(type, [&](const auto data_type_t) {
+    using ColumnDataType = typename decltype(data_type_t)::type;
+    new_segment = std::make_shared<ValueSegment<ColumnDataType>>();
+  });
+  return new_segment;
+}
+
+void Table::_append_column_to_chunks(const std::string& type) {
+  DebugAssert(row_count() == 0, "Cannot append new columns to already existing chunks.");
```

Comment:
```suggestion
```
We think it makes more sense to do the check as soon as possible and not deep down in the private methods.